dhis-2/dhis-services/dhis-service-analytics/src/main/java/org/hisp/dhis/analytics/DataQueryParams.java

```diff
@@ -690,6 +691,8 @@ protected QueryKey getQueryKey()
             .add( "order", order )
             .add( "timeField", timeField )
             .add( "orgUnitField", orgUnitField )
+            .add( "expressiondimensionitems", this.getExpressionDimensionItems().stream()
+                .map( edi -> ((ExpressionDimensionItem) edi).getExpression() ).collect( Collectors.joining() ) )
             .addIgnoreNull( "apiVersion", apiVersion )
             .addIgnoreNull( "locale", locale );
```

Comment:
Would be nice with another test in `QueryKeyTest` to validate that the hash key is different based on different EDIs.

Reply:
Done